### PR TITLE
refactored code from context based to api based

### PR DESCRIPTION
**Pull request by AnuragSahu11**
Refactored video library code from context based to API based. That's why number of files changed is so much.
The main changes are these files

files to check 
- **server requests.jsx** 
- **auth-context.jsx**  
- **auth-reducer.jsx**

changes in other files are very minor just to accomodate these changes.
Added mockbee watchlater files.
Added Video to the landing page

### CODE REVIEW

**Review by rishichourasia — APPROVED** [review]
Looks good to me , can be refactored more.

File: TradePlay/src/context/auth-context.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  useReducer,
+  useState,
+} from "react";
+import { userDataReducer } from "./auth-reducer";
+
+const AuthContext = createContext();
+const useAuth = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }) => {
+  const initialUserData = {
```
Comment (saadahmadhere) [inline TradePlay/src/context/auth-context.jsx:14]: Suggestion: you can pass this `initialUserData` object as a default state parameter to the reducer, and return the state in the default case of switch case.

**Review by saadahmadhere — COMMENTED** [review]
Great work chiseling your code, looks compact and more readable.

File: TradePlay/src/context/auth-context.jsx (hunk 1 above)
Comment (AnuragSahu11) [reply TradePlay/src/context/auth-context.jsx:14]: Yes, will do that

**Pull request merged**